src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -651,14 +652,22 @@ private void mapLocation(List<CompletableFuture<?>> futures, Location location,
   }
 
   private void mapVendorDetail(List<CompletableFuture<?>> futures, VendorDetail vendorDetail, Document doc) {
-    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_INSTRUCTIONS))
-      .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> vendorDetail.setInstructions((String) o))
-        .exceptionally(Mapper::logException)));
-
     Optional.ofNullable(mappings.get(Mapping.Field.NOTE_FROM_VENDOR))
+    .ifPresent(field -> futures.add(field.resolve(doc)
+      .thenAccept(o -> vendorDetail.setNoteFromVendor((String) o))
+      .exceptionally(Mapper::logException)));
+    
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_INSTRUCTIONS))
       .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> vendorDetail.setNoteFromVendor((String) o))
+        .thenAccept(o -> {
+        	String vendorCode = vendorDetail.getNoteFromVendor();       	
```

Comment:
As these are futures, there is no guarantee that the NOTE_FROM_VENDOR code above is always executed before this line. This code must move inside the NOTE_FROM_VENDOR block if we have to set this field conditionally.

Reply:
yeah... Earlier, when I pushed changes, Jenkins build fine but my local build kept failing randomly. I was wondering something was missing until your comment. I have refactored the code to move it inside `NOTE_FROM_VENDOR`. Looks good now. Thanks for catching it!

Reply:
is vendor_instructions supposed to be set only if NOTE_FROM_VENDOR? I just looked at the JIRA bug and doesn't look like it. Questions here is In the existing code Vendor code is already set to NOTE_FROM_VENDOR. If it is already present in the field, should it also go into another field VENDOR_INSTRUCTIONS?
If yes, then this code needs to be changed to use combinator. The way it is changed in line 661, the //orderNotes will be dropped if //purchaseOption/vendorCode is not present

Reply:
If //purchaseOption/vendorCode is not present, we want to keep //orderNotes as is. Quote from jira; I think its implied here `If there is already data in the Instructions to Vendor field (which would have come from the GobiAPI OrderNotes field), place the 5 character code before that data, followed by a space, e.g. EBS3U Please rush.`

Reply:
BTW, from what I understand, //purchaseOption/vendorCode will always be there

Reply:
> If //purchaseOption/vendorCode is not present, we want to keep //orderNotes as is. Quote from jira; I think its implied here `If there is already data in the Instructions to Vendor field (which would have come from the GobiAPI OrderNotes field), place the 5 character code before that data, followed by a space, e.g. EBS3U Please rush.`

correct we want to keep the //orderNotes irrespective of /vendorcode. The code here doesn't fo that. If //orderNotes is not present/Null. (which is mapped to NOTE_FROM_VENDOR). It doesn't go into the inner block and thus making VENDOR_INSTRUCTIONS null.

Reply:
> BTW, from what I understand, //purchaseOption/vendorCode will always be there

I don't think it's a mandatory field, we have few examples (PO_listed_print_monograph) where the //purchaseOption/vendorCode field doesn't come through.